api/files/views.py

```diff
@@ -72,6 +76,7 @@ class FileDetail(JSONAPIBaseView, generics.RetrieveUpdateAPIView, FileMixin):
         CheckedOutOrAdmin,
         base_permissions.TokenHasScope,
         PermissionWithGetter(ContributorOrPublic, 'target'),
+        RegistrationFileDetailPermission,
```

Comment:
If the permission class *is* needed, then it's a pure replacement for the `PermissionWithGetter(...)` above, and that line can be removed.